src/proj.h

```diff
@@ -243,58 +258,79 @@ typedef struct projCtx_t PJ_CONTEXT;
 typedef int *PJ_FILE;
 
 
+/* Functionality for handling thread contexts */
+PJ_CONTEXT *proj_context_create (char *setup, ...);
+void proj_context_set (PJ *P, PJ_CONTEXT *ctx);
+void proj_context_inherit (PJ *parent, PJ *child);
+void proj_context_destroy    (PJ_CONTEXT *ctx);
+
+
 /* Manage the transformation definition object PJ */
-PJ  *pj_create (const char *definition);
-PJ  *pj_create_argv (int argc, char **argv);
-void pj_free (PJ *P);
+PJ  *proj_create (PJ_CONTEXT *ctx, const char *definition);
+PJ  *proj_create_argv (PJ_CONTEXT *ctx, int argc, char **argv);
+PJ  *proj_create_pipeline (PJ_CONTEXT *ctx, const char *def_left, const char *def_right);
```

Comment:
I thought pipelines where multi-stages transform. left and right seems to be a limitation to 2